### PR TITLE
Return structured metadata in resources apis

### PR DESCRIPTION
**Pull request by cloudinaraz**
#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [x] New feature
- [ ] Bug fix
- [ ] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all of the tests pass.

### CODE REVIEW

File: tests/ApiTest.php [hunk 1]
```diff
@@ -155,6 +161,44 @@ protected static function delete_streaming_profiles($api)
...
+                $api->add_metadata_field(
+                    array(
+                        "external_id" => self::$api_test_metadata_field,
+                        "label" => self::$api_test_metadata_field,
+                        "type" => "string",
+                        "mandatory" => "true",
+                        "default_value" => UNIQUE_TEST_METADATA_FIELD_DEFAULT_VALUE
+                    )
+                );
+            } catch (Exception $e) {
```
Comment (const-cloudinary) [inline tests/ApiTest.php:198]: If this fails, all tests relying on metadata will obviously fail, and we will not know why. We don't need to catch exception here, if it fails, it fails.

File: tests/ApiTest.php [hunk 2]
```diff
@@ -56,18 +58,21 @@ public static function setUpBeforeClass()
...
+            $api = new Cloudinary\Api();
+
             self::$api_test_tag = UNIQUE_TEST_TAG;
 
-            self::$api_test = "api_test" . SUFFIX;
-            self::$api_test_2 = "api_test2" . SUFFIX;
-            self::$api_test_3 = "api_test,3" . SUFFIX;
-            self::$api_test_4 = "api_test4" . SUFFIX;
-            self::$api_test_5 = "api_test5" . SUFFIX;
+            self::$api_test = API_TEST_PREFIX . SUFFIX;
```
Comment (const-cloudinary) [inline tests/ApiTest.php:65]: Nice!

File: tests/ApiTest.php [hunk 3]
```diff
@@ -155,6 +161,44 @@ protected static function delete_streaming_profiles($api)
...
+         * Delete the metadata field created for test purposes
+         *
+         * @param \Cloudinary\Api $api
+         */
+        protected static function delete_metadata_field($api)
+        {
+            $metadata_field = self::$api_test_metadata_field;
+            try {
+                $api->delete_metadata_field($metadata_field);
+            } catch (Exception $e) {
```
Comment (const-cloudinary) [inline tests/ApiTest.php:175]: On teardown, we can log error here.

**Review by const-cloudinary — CHANGES_REQUESTED** [review]
@cloudinaraz , very good PR! 

Please fix a few minor issues.

File: tests/ApiTest.php (hunk 2 above)
Comment (cloudinaraz) [reply tests/ApiTest.php:65]: Thanks @const-cloudinary! I appreciate it :)

**Pull request merged**